packages/wmr/src/lib/prerender.js

```diff
@@ -96,8 +96,33 @@ async function workerCode({ cwd, out, publicPath, customRoutes }) {
 	let head = { lang: '', title: '', elements: new Set() };
 	globalThis.wmr = { ssr: { head } };
 
+	// Define `_fetch` to avoid infinite loop in globalThis.fetch.
+	let _fetch;
+	const nodeVersion = process.version;
+	const isNodeVersionOver18 = Number(nodeVersion.split('.')[0].slice(1)) >= 18;
+	// Native fetch is only available in Node.js 18 later.
+	if (isNodeVersionOver18) {
+		_fetch = fetch;
+	}
+	// @ts-ignore
+	delete globalThis.fetch;
 	// @ts-ignore
-	globalThis.fetch = async url => {
+	globalThis.fetch = async (url, options) => {
+		const pattern = /^https?:\/\/[\w/:%#\\$&\\?\\(\\)~\\.=\\+\\-]+/;
+		const isExternalUrl = pattern.test(String(url));
+		if (isExternalUrl) {
+			// Use native fetch if Node.js version 18 later when `prerender`
+			if (isNodeVersionOver18) {
+				return _fetch(url, options).then(res => ({
+					text: () => res.text(),
+					json: () => res.text().then(JSON.parse)
+				}));
```

Comment:
We don't need to limit this to Node 18.

While indeed, `fetch()` won't work automatically, users can polyfill it in older versions of Node. We only really care about a `fetch()` existing, not what version of Node they're using (though it's a good idea to warn if it's `undefined` and their version is older than 18)